### DISCUSSION TITLE
Loading pages from collections dynamically

### DISCUSSION THREAD

**Discussion by dijkr** [statamic/cms#8501] [Q&A] [answered]
Before using Statamic, I was using Blade views. And was getting the data from the database directly. Now I'd rebuild all views to work with Antlers and so far its great. What I just can't solve yet, is how to dynamically load a page, bases on the slug. I got a category-collection with pages. Every page is another category. 

```
{{ collection:promotions }}
        <div class="productbanner">
            <div class="productbanner-item">
                <img src="{{ banner }}">
                <div class="productbanner-text">
                    <h1>{{ category.name }}</h1>
                </div>
            </div>
        </div>
{{ /collection:promotions }}
```
This will display every page, but I would like to only display the page for the category. For example the URL will match the slug of the page. Could this be a condition to just load that page? Or would there be a better option to solve this? The idea is that a webshop has many categories and the editors would like to change the banner for the category themselves. 

Probably the docs will explain something about my question, but I can't really solve it yet. If the docs are explaining this, where would that be? It would be nice if I could fix issues like this myself, with the help of the docs. :)

**Comment by jasonvarga** [comment, marked as answer]
If I'm understanding, you are on a page like `/foo/bar` and you want to filter that collection by the slug (`bar`, maybe).

You can do something like this:

```
{{ collection:promotions :slug:is="segment_2" }}
```

**Reply by dijkr** [reply]
Thank you very much! This does exactly what I was looking for. Does segment_2 referring to the second part of the URL? Is there any documentation what the logic is behind this?

**Reply by jasonvarga** [reply]
Yep its the second part of the url. https://statamic.dev/variables/segment_x
